apps/nextra/pages/en/build/sdks/ts-sdk/building-transactions.mdx

```diff
@@ -0,0 +1,201 @@
+	  function: "0x1::aptos_account::transfer",
+	  functionArguments: [destination.accountAddress, 100],
+  },
+});
+```
+
+<Callout type="info">
+There is a more advanced format to pass in `functionArguments` called [Binary Canonical Serialization (BCS)](./building-transactions/bcs-format.mdx) format which is how the Aptos chain parses function arguments. The SDK converts TypeScript primitives to BCS format behind the scenes via an API call.
+</Callout>
+
+#### Building Options
+
+You can customize the way your transaction executes by passing in `options: {...}` when building. Some of the most commonly used options are:
+
+1. `maxGasAmount` - This caps the amount of gas you are willing to pay for to execute this transaction.
+2. `gasUnitPrice` - You can specify a higher than minimum price per gas to be executed with higher priority by the Aptos network.
+3. `expireTimestamp` - This gives a concrete time the transaction must execute by or it will be canceled.
+
+The SDK provides sensible defaults for these values if they are not specified explicitly.
+
+### Simulate (Optional)
+
+Every transaction on the Aptos chain has a gas fee associated with how much work the network machines have to do when executing the transaction. In order to estimate the cost associated with that, you can simulate transactions before committing them. 
+
+<Callout type="info">
+This simulation only requires the `publicKey` of an account since it will not impact the actual state of the ledger.
+</Callout>
+
+You can execute the simulation by using `aptos.transaction.simulate.simple(...)` like so:
+
+```tsx filename="build-a-transaction.ts"
+const [userTransactionResponse] = await aptos.transaction.simulate.simple({
+  signerPublicKey: signer.publicKey,
+  transaction,
+});
+// If the fee looks ok, continue to signing!
+```
+
+### Sign
+
+Once the transaction is built and the fees seem reasonable, you can sign the transaction with `aptos.transaction.sign`. The signature must come from the `sender` account.
+
+```tsx filename="build-a-transaction.ts"
+// 3. Sign
+const senderAuthenticator = aptos.transaction.sign({
+  signer: sender,
+  transaction,
+});
+```
+
+### Submit
+
+Now that the transaction is signed, you can submit it to the network using `aptos.transaction.submit.simple` like so:
+
+```tsx filename="build-a-transaction.ts"
+// 4. Submit
+const committedTransaction = await aptos.transaction.submit.simple({
+  transaction,
+  senderAuthenticator,
+});
+```
+
+### Wait
+
+Finally, you can wait for the result of the transaction by using [`aptos.waitForTransaction`](https://aptos-labs.github.io/aptos-ts-sdk/@aptos-labs/ts-sdk-1.12.2/classes/Aptos.html#waitForTransaction) and specifying the hash of the transaction you just submitted like so:
+
+```tsx filename="build-a-transaction.ts"
+// 5. Wait
+const executedTransaction = await aptos.waitForTransaction({ transactionHash: committedTransaction.hash });
+```
+</Steps>
```

Comment:
Note after this, we'll have to probably mention the long wait for transaction somewhere.  Though, I'm also perfectly fine with it not being well documented.

Reply:
What is the long wait? (When I was trying things out it seemed to return in a couple seconds?)

Reply:
It's a specific optimization that was added in the last release, I'll message you about it